### PR TITLE
Fix X-Forwarded-For in VIP_Request_Block

### PR DESCRIPTION
**Pull request by sjinks**
## Description

The original implementation of `VIP_Request_Block::ip()` incorrectly handles the `X-Forwarded-For` header, which sometimes leads to up to 2,625 additional IPs being blocked. This PR attempts to fix this issue.

For example, `VIP_Request_Block::ip( '1.1.1.1' )` could result in `11.1.1.1`, `21.1.1.1`,…, `251.1.1.199` being blocked.

As a side effect, this PR makes the `VIP_Request_Block` class more testable.

### Plugin Updated: VIP Request Block

The plugin now correctly handles partial IP matches in the X-Forwarded-For header.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test

I have added a test for my bug fix. The CI should pass.

### CODE REVIEW

**Review by pschoffer — APPROVED** [review]
Thank you for the tests part of it as well!

**Comment by sjinks** [comment]
r2136-stacks

File: lib/class-vip-request-block.php [hunk 1]
```diff
@@ -41,32 +45,36 @@ public static function ip( string $value ) {
...
+		return false;
 	}
 
 	/**
 	 * Block by exact match for an arbitrary header.
 	 *
 	 * @param string $header HTTP header.
 	 * @param string $value header value.
-	 * @return void
+	 * @return void|bool
```
Comment (GaryJones) [inline lib/class-vip-request-block.php:77]: Seems like these functions can't return void anymore?

Comment (sjinks) [reply lib/class-vip-request-block.php:77]: @GaryJones they can, please check https://github.com/Automattic/vip-go-mu-plugins/pull/2755/files/49a367ea0b1e502ee6c3de2394bad6ce1b25856a#diff-2aaa18fe2bb071a15fb049eb6bc4dcf92755b5f1baa6171d13e26330f629b38fR108

These functions return `true` (or sometimes `false`) only when we run tests; in the production environment, `block_and_log()` will call `exit()`.

**Comment by sjinks** [comment]
v20211228.1

**Pull request merged**